UndertaleModLib/Models/UndertaleSprite.cs

```diff
@@ -42,23 +42,39 @@ public void Serialize(UndertaleWriter writer)
     {
         writer.Write(PageWidth);
         writer.Write(PageHeight);
-        writer.Write(TexBlob.Length);
-        writer.Write(TexBlob);
+        // TODO: check if this is also the case on earlier versions
+        if (writer.undertaleData.IsVersionAtLeast(2023, 4)) {
+            // TODO: figure out what these bytes mean
+            writer.Write(0);
+        } else {
+            writer.Write(TexBlob.Length);
+            writer.Write(TexBlob);
+        }
     }
 
     /// <inheritdoc />
     public void Unserialize(UndertaleReader reader)
     {
         PageWidth = reader.ReadInt32();
         PageHeight = reader.ReadInt32();
-        TexBlob = reader.ReadBytes(reader.ReadInt32());
+        // TODO: check if this is also the case on earlier versions
+        if (reader.undertaleData.IsVersionAtLeast(2023, 4)) {
+            // TODO: figure out what these bytes mean
+            reader.ReadInt32();
```

Comment:
Usually, if there's unknown data, it's still good to store it so it can be written back on save (call it `Unknown` or something). That way, the tool is able to write the data file back byte for byte.